docs/guides/architecture/configuration-files.md

```diff
@@ -4,19 +4,82 @@ sidebar_position: 1
 
 ## `tauri.conf.json`
 
 This is the file used by the Tauri process. You can define the build settings (such as the [command run prior to `tauri build`][before-build-command] or [`tauri dev`][before-dev-command]), set the [name and version of your app][package-config], [control the Tauri process][tauri-config], and any plugin settings. Find all of the options for this config file in the [`tauri.conf.json` API reference].
 
 ## `Cargo.toml`
 
-Cargo's manifest file. You can declare Rust crates your app depends on, metadata about your app, and much more. For the full reference see Cargo's Manifest Format.
+Cargo's manifest file. You can declare Rust crates your app depends on, metadata about your app, and much more. If you do not intend to do a lot of backend development using Rust for your app then you probably won't be changing it very much, but it's important to know it exists and what it does.
+
+An example of a barebones `Cargo.toml` file for a Tauri project might look a little something like this:
+
+```toml
+[package]
+name = "app"
+version = "0.1.0"
+description = "A Tauri App"
+authors = ["you"]
+license = ""
+repository = ""
+default-run = "app"
+edition = "2021"
+rust-version = "1.57"
+
+[build-dependencies]
+tauri-build = { version = "1.0.0" }
+
+[dependencies]
+serde_json = "1.0"
+serde = { version = "1.0", features = ["derive"] }
+tauri = { version = "1.0.0", features = [ "api-all" ] }
+
+[features]
+# by default Tauri runs in production mode
+# when `tauri dev` runs it is executed with `cargo run --no-default-features` if `devPath` is an URL
+default = [ "custom-protocol" ]
+# this feature is used used for production builds where `devPath` points to the filesystem
+# DO NOT remove this
+custom-protocol = [ "tauri/custom-protocol" ]
+```
+
+The most important parts to take note of here are the `tauri-build` and `tauri` dependencies. Their versions must be the same as the version of your CLI in order to ensure that there won't be any issues running your app. If any of those three are not using the same version number you are highly likely to encounter issues.
+
+Cargo version numbers uses Semver versioning, and running `cargo update` will pull the latest available Semver compatible versions of all dependencies. What this means is that even if you specify `1.0.0` as the version for `tauri-build`, Cargo can actually decide to download version `1.0.4`, because it is the latest Semver compatible version available. According to Semver projects are required to update their major version number whenever a breaking change is introduced, meaning you should always be capable of safely upgrading to the latest minor and patch versions without fear of your code breaking.
```

Comment:
```suggestion
Cargo version numbers use [Semantic Versioning], and running `cargo update` will pull the latest available Semver-compatible versions of all dependencies. What this means is that even if you specify `1.0.0` as the version for `tauri-build`, Cargo can actually decide to download version `1.0.4`, because it is the latest Semver-compatible version available. According to Semver projects are required to update their major version number whenever a breaking change is introduced, meaning you should always be capable of safely upgrading to the latest minor and patch versions without fear of your code breaking.
```